src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -110,12 +113,49 @@ private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearch
                 throw new BadSearchRequest(errMsg);
             }
             if (response.searchResult != null) {
+                buildCaseTypesResults(response, caseTypeResults, crossCaseTypeSearchRequest);
                 caseDetails.addAll(searchResultToCaseList(response.searchResult));
                 totalHits += response.searchResult.getTotal();
             }
         }
 
-        return new CaseSearchResult(totalHits, caseDetails);
+        return new CaseSearchResult(caseTypeResults, totalHits, caseDetails);
+    }
+
+    private void buildCaseTypesResults(
+        MultiSearchResult.MultiSearchResponse response,
+        List<CaseTypeResults> caseFieldsAggregations,
+        CrossCaseTypeSearchRequest crossCaseTypeSearchRequest) {
+
+        if (hitsIsNotEmpty(response)) {
+            String indexName = getIndexName(response);
+            caseFieldsAggregations.add(new CaseTypeResults(getCaseTypeIDFromIndex(indexName,
+                crossCaseTypeSearchRequest.getCaseTypeIds()),
+                response.searchResult.getTotal())
+            );
+        }
+    }
+
+    private String getIndexName(MultiSearchResult.MultiSearchResponse response) {
+        return response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits")
+            .getAsJsonArray().get(0).getAsJsonObject().get("_index").toString();
+    }
+
+    private boolean hitsIsNotEmpty(MultiSearchResult.MultiSearchResponse response) {
+        return response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits").getAsJsonArray().size() != 0;
+    }
+
+    private String getCaseTypeIDFromIndex(final String index, List<String> caseTypeIds) {
+        String pureIndexFormat = applicationParams.getCasesIndexNameFormat().substring(2);
```

Comment:
Sorry I know I suggested using the property value! I think if doing so we need to be a bit smarter though e.g. what if the property were changed to `new_%s_cases`, then `substring` wouldn't work. Something to consider!

Reply:
Yes good ponts , but if we new_%s_cases that can lead in changing all index in CCD.

Reply:
I thought the same. I'll fix this